Add week 3 learning outcomes (#9)
foundersandcoders/coursebook: learning-outcomes.md

# Week 3 Learning Outcomes

## Testing

- [ ] **We can test the functionality of our app**
- [ ] We can write tests to mimic the behaviour of a user performing different actions
- [ ] We can check that passing a given input into our tests returns the expected output

### Modular functions

- [ ] **We can write testable, modular functions**
- [ ] We can write functions that add DOM nodes
- [ ] We can write functions that remove DOM nodes

> - [ ] We can write functions that modify DOM nodes

Reviewer: Why the specific focus on DOM nodes in the modular functions section?

Author: I was thinking of the to-do list project. Perhaps these would be more suited to learning outcomes for the project though, rather than the week.

Reviewer: I don't think those are bad outcomes for this week, I just don't know if they fit under "modular functions"

Author: I have changed the heading from 'Modular functions' to just 'Functions' to encompass all of these LOs

Reviewer: We have a general "Javascript" heading in some weeks. Could we use that here and move the scope LO under it too?

### Event listeners

> - [ ] **We can apply click event listeners to buttons**

Reviewer: This might be too specific? They should probably be using a form `"submit"` event for the project this week for example

- [ ] We can pass in named functions as the second argument

### Scope

> - [ ] **We can apply the concept of ‘scoping’ to functions and variables**

Reviewer: "apply the concept" might be a little vague/confusing. Can we think of a concrete application for scoping? Like "We can use function scope to..."

Author: How about "We can use function and block scope to store values inside temporary variables"?

I agree, the more concrete examples we can scaffold these LOs around, the better

- [ ] We can distinguish between block, function, and global scope so that we can control access to variables from different parts of our project

---

## Design

> - [ ] **We can create a clean colour palette for our project**

Reviewer: I'm not sure "clean" is a very measurable outcome 😆 
How about
```suggestion
- [ ] **We can create an attractive and accessible colour palette for our project**
```

- [ ] We can apply colours to HTML elements using CSS variables

---

## Developer Toolkit

> - [ ] **We can run automated tests in the browser (to check that our client side code works correctly)**

Reviewer: This isn't really a dev tooling outcome (other than that the tests will use `console.log`)
